### PR TITLE
CRM Improvements

### PR DESCRIPTION
**Pull request by KornelijusGlinskas**
TO DO: 

- [x] Preloader
- [x] Connect all dynamic data from api
- [x] Enable search
- [x] Enable filters

### CODE REVIEW

File: assets/js/components/admin/Reservations.jsx [hunk 1]
```diff
@@ -1,48 +1,171 @@
...
+      <i className="icon-cross" />
+      {text}
+    </button>
+  );
+
+  const isAnyFilterSelected = () => {
+    if (isCancelledFilter || isPendingFilter || isConfirmedFilter || isExpiredFilter) {
+      return true;
+    }
+    return false;
```
Comment (tadas-nfq) [inline assets/js/components/admin/Reservations.jsx]: I think you can shorten this to:
`return isCancelledFilter || isPendingFilter || isConfirmedFilter || isExpiredFilter;`

File: templates/contractor/index.html.twig [hunk 2]
```diff
@@ -3,9 +3,12 @@
 {% block title %} {{ "contractor_page.welcome"|trans({'%name%': app.user.firstname}, "messages") }}{% endblock %}
 
 {% block body %}
-    <div id="admin" data-key="{{ app.user.verificationKey }}">
+    
+    {# {% include('register-modal.html') with {'text': 'Add existing reservation'} %} #}
```
Comment (tadas-nfq) [inline templates/contractor/index.html.twig]: Remove commented code

File: templates/contractor/index.html.twig [hunk 3]
```diff
@@ -3,9 +3,12 @@
...
+    <div id="admin" data-key="{{ app.user.verificationKey }}" data-username="{{ app.user.username }}">
     </div> 
     {% block javascripts %}
         {{ encore_entry_script_tags('admin') }}
+        {# {{ encore_entry_script_tags('datepicker') }} #}
```
Comment (tadas-nfq) [inline templates/contractor/index.html.twig]: Remove commented code

File: webpack.config.js [hunk 4]
```diff
@@ -2,72 +2,66 @@ var Encore = require('@symfony/webpack-encore');
...
 
-    // enables Sass/SCSS support
-    .enableSassLoader()
-    .enableReactPreset()
+  // enables Sass/SCSS support
+  .enableSassLoader()
+  .enableReactPreset()
 
-    //.addEntry('admin', './assets/js/admin.js')
+  //.addEntry('admin', './assets/js/admin.js')
```
Comment (tadas-nfq) [inline webpack.config.js]: Remove commented code

File: assets/js/components/admin/Reservations.jsx (hunk 1 above)
Comment (KornelijusGlinskas) [reply assets/js/components/admin/Reservations.jsx]: Indeed! Thanks for good notice :)
